tests/phpunit/tests/media.php

```diff
@@ -3948,6 +3948,105 @@ public function data_special_contexts_for_the_content() {
 		);
 	}
 
+	/**
+	 * Tests that the content media count is not affected by `the_excerpt()` calls for posts that contain images.
+	 *
+	 * @ticket 56588
+	 *
+	 * @covers ::the_excerpt
+	 * @covers ::wp_trim_excerpt
+	 */
+	public function test_the_excerpt_does_not_affect_content_media_count() {
+		global $wp_query, $wp_the_query;
+
+		/*
+		 * Use the filter to alter the threshold for not lazy-loading to the first 2 elements,
+		 * then use a post that contains exactly 2 images.
+		 */
+		$this->force_omit_loading_attr_threshold( 2 );
+		$post_content  = '<img src="example.jpg" width="800" height="600">';
+		$post_content .= '<p>Some text.</p>';
+		$post_content .= '<img src="example2.jpg" width="800" height="600">';
+
+		$post_id = self::factory()->post->create(
+			array(
+				'post_content' => $post_content,
+				'post_excerpt' => '',
+			)
+		);
+
+		$wp_query     = new WP_Query( array( 'post__in' => array( $post_id ) ) );
+		$wp_the_query = $wp_query;
+		$this->reset_content_media_count();
```

Comment:
@joemcgill What do you mean by fixture? You mean in a `set_up()` method for example?

Indeed this isn't automatically reset since it relies on a static variable inside the function.

Reply:
Yup, that what I meant! If the count is not reset before each method that tests image output, then the unit tests themselves can cause side effects to other tests. Looks like we're already doing this manually in several methods, so moving it to a shared fixture would make the whole test class a bit cleaner and more predictable.

Reply:
Updated in https://github.com/WordPress/wordpress-develop/pull/4439/commits/7f41cb63d2db2572de8829c3614ff8ed051f0247